Include/ceval.h

```diff
@@ -69,10 +69,10 @@ PyAPI_FUNC(void) Py_LeaveRecursiveCall(void);
 
 #define Py_ALLOW_RECURSION \
   do { unsigned char _old = PyThreadState_GET()->recursion_critical;\
```

Comment:
`recursion_critical` needs fixing up here too. Are you sure you grepped the whole codebase?

Reply:
How did this even compile and pass CI?

Reply:
I suppose this macro is only used in third-party code? Which of course makes wonder whether it still works as expected.